// components/DescriptionComponent/DescriptionComponent.js
import React from "react"
import PropTypes from "prop-types"
import "./DescriptionComponent.sass"
import {Container, Row, Col} from 'react-bootstrap'

export const DescriptionComponent = ({title, subTitle, image, mainText, sideComponent}) => {
  return (
    <div className="description-container-component">
      <Container>
        {!title || (
          <Row>
            <Col>
              <h1 className="title">{title}</h1>
              {!subTitle || <h5>{subTitle}</h5>}
            </Col>
          </Row>
        )}
        {!image || (
          <Row>
            <Col className="description-container-image-container">
              <img className="description-container-image" alt="Header" src={image} />
            </Col>
          </Row>
        )}
        <Row>
          <Col lg={sideComponent ? 8 :  12}>
            <p className="description-container-text">{mainText}</p>
          </Col>
          {!sideComponent || (
            <Col lg={4}>
              <p className="description-container-text">{sideComponent.content}</p>
            </Col>
          )}
        </Row>
      </Container>
    </div>
  )
}

DescriptionComponent.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  mainText: PropTypes.string,
  subTitle: PropTypes.string,
  sideComponent: PropTypes.object
}
